src/main/java/com/notably/commons/compiler/parser/block/Block.java

```diff
@@ -0,0 +1,95 @@
+package com.notably.commons.compiler.parser.block;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a node in a Markdown Abstract Syntax Tree (AST).
+ * Read more: https://github.github.com/gfm/#appendix-a-parsing-strategy
+ */
+public abstract class Block {
```

Comment:
Just a thought, maybe it would be confusing to have another class named `Block`.
Although they are packaged differently, not sure how it would affect auto import..
Maybe we could change it to something similar `Section` etc ?